openpype/hosts/blender/plugins/create/create_workfile.py

```diff
@@ -0,0 +1,110 @@
+import bpy
+
+from openpype.pipeline import CreatedInstance, AutoCreator
+from openpype.client import get_asset_by_name
+from openpype.hosts.blender.api.plugin import BaseCreator
+from openpype.hosts.blender.api.pipeline import (
+    AVALON_PROPERTY,
+    AVALON_CONTAINERS
+)
+
+
+class CreateWorkfile(BaseCreator, AutoCreator):
+    """Workfile auto-creator.
+
+    The workfile instance stores its data on the `AVALON_CONTAINERS` collection
+    as custom attributes, because unlike other instances it doesn't have an
+    instance node of its own.
+
+    """
+    identifier = "io.openpype.creators.blender.workfile"
+    label = "Workfile"
+    family = "workfile"
+    icon = "fa5.file"
+
+    def create(self):
+        """Create workfile instances."""
+        current_instance = next(
+            (
+                instance for instance in self.create_context.instances
+                if instance.creator_identifier == self.identifier
+            ),
+            None,
+        )
+
+        project_name = self.project_name
+        asset_name = self.create_context.get_current_asset_name()
+        task_name = self.create_context.get_current_task_name()
+        host_name = self.create_context.host_name
+
+        if not current_instance:
+            asset_doc = get_asset_by_name(project_name, asset_name)
+            subset_name = self.get_subset_name(
+                task_name, task_name, asset_doc, project_name, host_name
+            )
+            data = {
+                "asset": asset_name,
+                "task": task_name,
+                "variant": task_name,
+            }
+            data.update(
+                self.get_dynamic_data(
+                    task_name,
+                    task_name,
+                    asset_doc,
+                    project_name,
+                    host_name,
+                    current_instance,
+                )
+            )
+            self.log.info("Auto-creating workfile instance...")
+            current_instance = CreatedInstance(
+                self.family, subset_name, data, self
+            )
+            instance_node = bpy.data.collections.get(AVALON_CONTAINERS, {})
+            current_instance.transient_data["instance_node"] = instance_node
+            self._add_instance_to_context(current_instance)
+        elif (
+            current_instance["asset"] != asset_name
+            or current_instance["task"] != task_name
+        ):
+            # Update instance context if it's different
+            asset_doc = get_asset_by_name(project_name, asset_name)
+            subset_name = self.get_subset_name(
+                task_name, task_name, asset_doc, project_name, host_name
+            )
+
+            current_instance.update(
+                {
+                    "asset": asset_name,
+                    "task": task_name,
+                    "subset": subset_name,
+                }
+            )
```

Comment:
This is causing issues because `current_instance` is not a `dict`!

```
Creator: Workfile (io.openpype.creators.blender.workfile)

Error: 'CreatedInstance' object has no attribute 'update'

Traceback (most recent call last):
  File "C:\Dev\Ynput\OpenPype\openpype\pipeline\create\context.py", line 2036, in _create_with_unified_error
    result = creator.create(*args, **kwargs)
  File "C:\Dev\Ynput\OpenPype\openpype\hosts\blender\plugins\create\create_workfile.py", line 77, in create
    current_instance.update(
AttributeError: 'CreatedInstance' object has no attribute 'update'
```

Reply:
Thanks, well spotted. Fixed with https://github.com/ynput/OpenPype/pull/5910/commits/45699444ef8d4e466d3fed4652d26b4d1c55033a

Try and break it again!